app/src/main/java/org/proninyaroslav/libretorrent/core/TorrentDownload.java

```diff
@@ -195,8 +195,7 @@ public void alert(Alert<?> alert)
 
     private boolean shouldPauseDownloads()
     {
-        IntentFilter intentFilter = new IntentFilter(Intent.ACTION_BATTERY_CHANGED);
-        batteryStatus = context.registerReceiver(null, intentFilter);
+        Intent batteryStatus = context.registerReceiver(null, intentFilter);
```

Comment:
I do not like that I have to re-initialize this every time this method is called but when I initialized it in the constructor, it retained the initial battery value it received when the torrent was added.